platform/commonUI/edit/src/actions/SaveAction.js

```diff
@@ -127,17 +118,13 @@ define(
             function doSave() {
                 //This is a new 'virtual object' that has not been persisted
                 // yet.
-                if (!domainObject.getModel().persisted){
+                if (domainObject.getModel().persisted === undefined){
                     return getParent(domainObject)
-                            .then(doWizardSave)
-                            .then(persistObject)
-                            .then(getParent)//Parent may have changed based
-                                            // on user selection
-                            .then(locateObjectInParent)
-                            .then(function(){
-                                return fetchObject(domainObject.getId());
-                            })
-                        .catch(doNothing);
+                        .then(doWizardSave)
+                        .then(getParent)
+                        .then(copyService.perform.bind(copyService, domainObject))
```

Comment:
I might be thinking too hard, but I believe this will duplicate linked subobjects when possible, as the copy service is a deep copy as per design of the Duplicate Action.  In this case, we'd want to prevent the copy service from duplicating subobjects that already exist.